File: 0062-unique-paths/0062-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[] dp = new int[n];
        dp[0] = 1;

        for (int row = 0; row < m; row++) {
            for (int col = 0; col < n; col++) {
                if (col > 0) {
                    dp[col] += dp[col - 1];
                }
            }
        }
        return dp[n - 1];
    }
}

File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [0] * n
        dp[0] = 1

        for _ in range(m):
            for col in range(n):
                if col > 0:
                    dp[col] += dp[col - 1]

        return dp[n - 1]
